src/runtime/pkg/device/drivers/vfio.go

```diff
@@ -211,20 +214,38 @@ func (device *VFIODevice) Load(ds config.DeviceState) {
 	device.GenericDevice.Load(ds)
 
 	for _, dev := range ds.VFIODevs {
-		var vfioDev config.VFIODev = config.VFIOPCIDev{
-			ID:       dev.ID,
-			Type:     config.VFIODeviceType(dev.Type),
-			BDF:      dev.BDF,
-			SysfsDev: dev.SysfsDev,
+		var vfio config.VFIODev
+
+		if (*device.VfioDevs[0]).GetType() == config.VFIOAPDeviceMediatedType {
```

Comment:
Should probably be a switch to catch bad input from the config. This is external input, so this really needs to be validated.

Reply:
Done. Thanks. :wink: